x-pack/plugins/saved_objects_tagging/public/components/edition_modal/create_or_edit_modal.tsx

```diff
@@ -0,0 +1,234 @@
+
+  // we don't want this value to change when the user edit the name.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const initialName = useMemo(() => tag.name, []);
+
+  const setName = useMemo(() => setField('name'), [setField]);
+  const setColor = useMemo(() => setField('color'), [setField]);
+  const setDescription = useMemo(() => setField('description'), [setField]);
+
+  const isEdit = useMemo(() => mode === 'edit', [mode]);
+
+  const previewTag: TagAttributes = useMemo(() => {
+    return {
+      ...tag,
+      name: tag.name || 'tag',
+      color: validateTagColor(tag.color) ? '#000000' : tag.color,
+    };
+  }, [tag]);
+
+  const onFormSubmit = useCallback(async () => {
+    if (!validation.valid) {
+      return;
+    }
+
+    setSubmitting(true);
+    await onSubmit();
+    setSubmitting(false);
+  }, [validation, onSubmit]);
+
+  return (
+    <EuiModal onClose={onClose}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          {isEdit ? (
+            <FormattedMessage
+              id="xpack.savedObjectsTagging.management.editModal.title"
+              defaultMessage="Edit '{name}' tag"
+              values={{
+                name: initialName,
+              }}
+            />
+          ) : (
+            <FormattedMessage
+              id="xpack.savedObjectsTagging.management.createModal.title"
+              defaultMessage="Create tag"
+            />
+          )}
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiForm data-test-subj="tagModalForm" role="form">
+          <EuiFlexGroup justifyContent="spaceBetween" alignItems="baseline">
+            <EuiFlexItem>
+              <EuiFormRow
+                data-test-subj="createModalRow-name"
+                fullWidth={true}
+                label={i18n.translate('xpack.savedObjectsTagging.tagAttributeLabels.name', {
+                  defaultMessage: 'Name',
+                })}
+                isInvalid={!!validation.errors.name}
+                error={validation.errors.name}
+              >
+                <EuiFieldText
+                  value={tag.name}
+                  onChange={(e) => setName(e.target.value)}
+                  data-test-subj="createModalField-name"
+                />
+              </EuiFormRow>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiFormRow
+                data-test-subj="createModalRow-color"
+                fullWidth={true}
+                label={i18n.translate('xpack.savedObjectsTagging.tagAttributeLabels.color', {
+                  defaultMessage: 'Color',
+                })}
+                isInvalid={!!validation.errors.color}
+                error={validation.errors.color}
+                labelAppend={
+                  <EuiButtonEmpty
+                    onClick={() => setColor(getRandomColor())}
+                    size="xs"
+                    style={{ height: '18px', fontSize: '0.75rem' }}
```

Comment:
So... EUI dirty secret: if you pass in an `onClick` and no `href` to `EuiLink`, it'll render as a `<button>` (and, vice versa, if you pass an `href` and no `onClick` to any of the `EuiButton*` components, it'll render as a `<a>`).

@ryankeairns Should we be opening an EUI issue for this?

Reply:
EUI has plans to address this by setting a smaller font size on small buttons. Given that, we can safely remove the `style` overrides here, live with the larger button font in the meantime, and rely on a future EUI upgrade to resolve this.

Regardless, thank you for the attention to detail!

Reply:
A couple of clarifications to my comment...
- This is being 'fixed' on the Amsterdam theme which is targeted as the default theme for 7.12
- It will likely only affect the `xs` size, which is what we have here so that's good

Reply:
7.12 is far. I will stick with the manual fix for now them. Will have no impact when switching theme, and we will remove the manual styles then.